keep/providers/netdata_provider/netdata_provider.py

```diff
@@ -0,0 +1,91 @@
+"""
+Netdata is a cloud-based monitoring tool that provides real-time monitoring of servers, applications, and devices.
+"""
+
+from typing import Optional
+
+from keep.api.models.alert import AlertDto, AlertStatus, AlertSeverity
+from keep.contextmanager.contextmanager import ContextManager
+from keep.providers.base.base_provider import BaseProvider
+from keep.providers.models.provider_config import ProviderConfig
+
+class NetdataProvider(BaseProvider):
+    """Get alerts from Netdata into Keep."""
+
+    webhook_description = ""
+    webhook_template = ""
+    webhook_markdown = """
+💡 For more details on how to configure Netdata to send alerts to Keep, see the [Keep documentation](https://docs.keephq.dev/providers/documentation/netdata-provider).
+
+To send alerts from Netdata to Keep, Use the following webhook url to configure Netdata send alerts to Keep:
+
+1. In Netdata, go to Space settings.
+2. Go to "Alerts & Notifications".
+3. Click on "Add configuration".
+4. Add "Webhook" as the notification method.
+5. Add a name to the configuration.
+6. Select Room(s) to apply the configuration.
+7. Select Notification(s) to apply the configuration.
+8. In the "Webhook URL" field, add {keep_webhook_api_url}.
+9. Add a request header with the key "x-api-key" and the value as {webhook_api_key}.
+10. Leave the Authentication as "No Authentication".
+11. Add the "Challenge secret" as "keep-netdata-webhook-integration".
+12. Save the configuration.
+"""
+
+    SEVERITIES_MAP = {
+        "warning": AlertSeverity.WARNING,
+        "info": AlertSeverity.INFO,
+        "critical": AlertSeverity.CRITICAL,
+    }
+
+    STATUS_MAP = {
+        "reachable": AlertStatus.RESOLVED,
+        "unreachable": AlertStatus.FIRING,
+    }
+
+    PROVIDER_DISPLAY_NAME = "Netdata"
+    PROVIDER_TAGS = ["alert"]
+
+    def __init__(
+        self, context_manager: ContextManager, provider_id: str, config: ProviderConfig
+    ):
+        super().__init__(context_manager, provider_id, config)
+
+    def validate_config(self):
+        """
+        Validates required configuration for Prometheus's provider.
+        """
+        # no config
+        pass
+
+    @staticmethod
+    def _format_alert(
```

Comment:
please split the parameter fetching into separate code lines since it can raise a lot of exceptions